### PR TITLE
To `substrate#polkadot-v0.9.11`

### PR DESCRIPTION
**Pull request by AurevoirXavier**

### CODE REVIEW

**Comment by AurevoirXavier** [comment]
Failed to compile. After merging #1018. @AsceticBear

**Comment by AsceticBear** [comment]
> Failed to compile. After merging #1018. @AsceticBear

Try clean and rebuild again?  No problem in my env.

**Comment by AsceticBear** [comment]
@AurevoirXavier CI failed.

**Comment by AurevoirXavier** [comment]
Storage prefix checker is broken on metadata-v14.

File: frame/fee-market/Cargo.toml [hunk 1]
```diff
@@ -13,24 +13,23 @@ version     = "2.7.2"
 # crates.io
 bitvec     = { version = "0.20", default-features = false, features = ["alloc"] }
 codec      = { package = "parity-scale-codec", version = "2.1.1", default-features = false }
```
Comment (xiaoch05) [inline frame/fee-market/Cargo.toml:15]: this version should also updated to 2.2?

File: primitives/relay/Cargo.toml [hunk 2]
```diff
@@ -11,18 +11,20 @@ version     = "2.7.2"
 
 [dependencies]
 # crates.io
-codec = { package = "parity-scale-codec", version = "2.1.1", default-features = false }
+codec      = { package = "parity-scale-codec", version = "2.1.1", default-features = false }
```
Comment (xiaoch05) [inline primitives/relay/Cargo.toml:14]: this version should also update to 2.2?

File: frame/dvm/src/lib.rs [hunk 3]
```diff
@@ -419,7 +421,7 @@ impl<T: Config> Pallet<T> {
 
 		Self::deposit_event(Event::Executed(
 			transaction.source,
-			contract_address.unwrap_or_default(),
+			dest.unwrap_or_default(),
```
Comment (xiaoch05) [inline frame/dvm/src/lib.rs:424]: why remove this contract_address? instead by dest?

File: node/runtime/pangoro/src/pangolin_messages.rs [hunk 4]
```diff
@@ -117,6 +118,8 @@ impl MessageBridge for WithPangolinMessageBridge {
 
 	fn bridged_balance_to_this_balance(
 		bridged_balance: pangolin_primitives::Balance,
+		// TODO: S2S
+		_bridged_to_this_conversion_rate_override: Option<FixedU128>,
```
Comment (AsceticBear) [inline node/runtime/pangoro/src/pangolin_messages.rs:122]: @xiaoch05 @fewensa Please help review this change.

**Review by xiaoch05 — CHANGES_REQUESTED** [review]

File: frame/dvm/src/lib.rs (hunk 3 above)
Comment (AsceticBear) [reply frame/dvm/src/lib.rs:424]: The dest can be either `created_address` when deploying a contract or `to` when making a call.

File: frame/staking/src/lib.rs [hunk 5]
```diff
@@ -445,17 +451,17 @@ pub mod pallet {
...
-		BondKton(KtonBalance<T>),
+		KtonBonded(AccountId<T>, KtonBalance<T>),
 
-		/// An account has unbonded this amount. \[amount, now\]
-		UnbondRing(RingBalance<T>, BlockNumberFor<T>),
-		/// An account has unbonded this amount. [amount, now\]
-		UnbondKton(KtonBalance<T>, BlockNumberFor<T>),
+		/// An account has unbonded this amount. \[amount\]
+		RingUnbonded(AccountId<T>, RingBalance<T>),
+		/// An account has unbonded this amount. \[amount\]
```
Comment (xiaoch05) [inline frame/staking/src/lib.rs:463]:
```suggestion
		/// An account has unbonded this amount. \[account, amount\]
```

File: frame/staking/src/lib.rs [hunk 6]
```diff
@@ -445,17 +451,17 @@ pub mod pallet {
 		///
 		/// NOTE: This event is only emitted when funds are bonded via a dispatchable. Notably,
 		/// it will not be emitted for staking rewards when they are added to stake.
-		BondRing(RingBalance<T>, TsInMs, TsInMs),
+		RingBonded(AccountId<T>, RingBalance<T>, TsInMs, TsInMs),
 		/// An account has bonded this amount. \[amount, start, end\]
```
Comment (xiaoch05) [inline frame/staking/src/lib.rs:455]:
```suggestion
		/// An account has bonded this amount. \[account, amount, start, end\]
```

**Review by AsceticBear — APPROVED** [review]
I have focused on checking everything related to DVM and the fee market.  LGTM.

File: node/primitives/bridge/src/lib.rs [hunk 7]
```diff
@@ -180,6 +190,16 @@ impl Chain for Pangolin {
...
+
+	// TODO: S2S
+	fn max_extrinsic_size() -> u32 {
+		todo!()
+	}
+
+	// TODO: S2S
+	fn max_extrinsic_weight() -> Weight {
+		todo!()
+	}
```
Comment (xiaoch05) [inline node/primitives/bridge/src/lib.rs:202]: what's the function of these two methods. Do we need to complete these todo right now or later?

**Review by xiaoch05 — CHANGES_REQUESTED** [review]

**Comment by AurevoirXavier** [comment]
Search `TODO: S2S` and get the companions from upstream. cc @AsceticBear  @xiaoch05

File: node/primitives/bridge/src/lib.rs [hunk 8]
```diff
@@ -145,8 +153,56 @@ pub const TO_PANGORO_LATEST_RECEIVED_NONCE_METHOD: &str =
 
 /// Name of the `PangoroFinalityApi::best_finalized` runtime method.
 pub const BEST_FINALIZED_PANGORO_HEADER_METHOD: &str = "PangoroFinalityApi_best_finalized";
+
+/// All Polkadot-like chains allow normal extrinsics to fill block up to 75 percent.
+///
+/// This is a copy-paste from the Polkadot repo's `polkadot-runtime-common` crate.
+const NORMAL_DISPATCH_RATIO: Perbill = Perbill::from_percent(75);
```
Comment (AsceticBear) [inline node/primitives/bridge/src/lib.rs:160]: @xiaoch05 I guess we can reuse already existed const variable here https://github.com/darwinia-network/darwinia-common/blob/main/node/runtime/common/src/lib.rs#L53-L56

**Comment by AurevoirXavier** [comment]
Ready to merge?

**Comment by xiaoch05** [comment]
> Ready to merge?

wait for a moment

File: node/primitives/bridge/Cargo.toml [hunk 9]
```diff
@@ -11,6 +11,7 @@ version     = "2.7.2"
 
 [dependencies]
 # darwinia-network
+common-runtime         = { default-features = false, path = "../../runtime/common" }
```
Comment (AurevoirXavier) [inline node/primitives/bridge/Cargo.toml:14]: Primitives should not depend on runtime.

Comment (xiaoch05) [reply node/primitives/bridge/Cargo.toml:14]: if so, we need define the types independently

Comment (AsceticBear) [reply node/primitives/bridge/Cargo.toml]: How about creating an issue firstly, we can find a way to fix it later. So we can merge this pr today.

Comment (xiaoch05) [reply node/primitives/bridge/Cargo.toml]: I find the next version of parity-bridges-common. This definitions are changed too?

**Review by xiaoch05 — APPROVED** [review]

**Review by AsceticBear — APPROVED** [review]
LGTM now

**Pull request merged**